In [1]:
#Import Python Packages
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import os
pd.options.display.float_format = '{:.2f}'.format
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score

**ACT Math 11**

In [2]:
#Loading the dataframe
filename = os.path.join(os.getcwd(), "dataframes", "actMath_11_encoded_df.csv")
actMath11_df = pd.read_csv(filename, header = 0)

In [3]:
actMath11_df.head()

studentId  is_proficient  proficient_score  proficient_diff  \
0      45054          False              0.53            -8.00   
1      45054          False              0.53            -8.00   
2      45440          False              0.94            -1.00   
3      45440          False              0.94            -1.00   
4      45440          False              0.94            -1.00   

   school_High School A  school_High School B  school_High School C  \
0                  True                 False                 False   
1                  True                 False                 False   
2                 False                 False                 False   
3                 False                 False                 False   
4                 False                 False                 False   

   school_High School D  iready_math  iready_reading  ...  \
0                 False            0               0  ...   
1                 False            0               0  ...   
2                 False            0               0  ...   
3                 False            0               0  ...   
4                 False            0               0  ...   

   category_PERFORMING_ARTS  category_VISUAL_ARTS  category_TECHNOLOGY  \
0                     False                 False                False   
1                     False                 False                False   
2                     False                 False                False   
3                     False                 False                False   
4                     False                 False                False   

   category_BUSINESS  category_TRADES  category_AGRICULTURE  \
0              False             True                 False   
1              False             True                 False   
2              False            False                 False   
3              False            False                 False   
4              False            False                 False   

   category_PHYSICAL_EDUCATION  category_HEALTH  category_CAREER_PREP  \
0                        False            False                 False   
1                        False            False                 False   
2                        False            False                 False   
3                        False            False                 False   
4                        False            False                 False   

   category_MILITARY_SCIENCE  
0                      False  
1                      False  
2                      False  
3                      False  
4                      False  

[5 rows x 26 columns]

In [4]:
actMath11_df.columns

Index(['studentId', 'is_proficient', 'proficient_score', 'proficient_diff',
       'school_High School A', 'school_High School B', 'school_High School C',
       'school_High School D', 'iready_math', 'iready_reading', 'bbb_literacy',
       'category_MATHEMATICS', 'category_ENGLISH', 'category_SCIENCE',
       'category_SOCIAL_STUDIES', 'category_FOREIGN_LANGUAGE',
       'category_PERFORMING_ARTS', 'category_VISUAL_ARTS',
       'category_TECHNOLOGY', 'category_BUSINESS', 'category_TRADES',
       'category_AGRICULTURE', 'category_PHYSICAL_EDUCATION',
       'category_HEALTH', 'category_CAREER_PREP', 'category_MILITARY_SCIENCE'],
      dtype='object')

In [5]:
#Create labeled examples from the dataset
#X = actMath11_df.drop(columns=['is_proficient', 'proficient_score', 'proficient_diff', 'school_High School A', 'school_High School B', 'school_High School C', 'school_High School D'], inplace=False)
X = actMath11_df[['category_MATHEMATICS', 'category_ENGLISH', 'category_SCIENCE',
       'category_SOCIAL_STUDIES', 'category_FOREIGN_LANGUAGE',
       'category_PERFORMING_ARTS', 'category_VISUAL_ARTS',
       'category_TECHNOLOGY', 'category_BUSINESS', 'category_TRADES',
       'category_AGRICULTURE', 'category_PHYSICAL_EDUCATION',
       'category_HEALTH', 'category_CAREER_PREP', 'category_MILITARY_SCIENCE']]
y = actMath11_df['proficient_score']

In [6]:
#Create training and test datasets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.15, random_state=42)

In [7]:
# Create the  LinearRegression model object 
model = LinearRegression()

# Fit the model to the training data 
model.fit(X_train, y_train)

#  Make predictions on the test data 
prediction = model.predict(X_test)

In [8]:
# The mean squared error
print('\nModel Performance\n\nRMSE =   %.2f'
      % np.sqrt(mean_squared_error(y_test, prediction)))
# The coefficient of determination: 1 is perfect prediction
print(' R^2 =   %.2f'
      % r2_score(y_test, prediction))


Model Performance

RMSE =   0.17
 R^2 =   0.25


Model Performance When I included all columns as features:
RMSE = 0.17
R^2 = 0.25

I got the same results above when I use only courses and only schools

Model Performance When only using vendors as features:
RMSE = 0.19
R^2 = -0.00